dvc/objects/file.py

```diff
@@ -28,8 +28,11 @@ def __init__(
 
     @property
     def size(self):
-        if not (self.path_info and self.fs):
+        if self.hash_info.size is not None:
+            return self.hash_info.size
+        if not (self.fs and self.path_info):
             return None
+
         return self.fs.getsize(self.path_info)
```

Comment:
Very good point about `getsize`. Indeed, `hash_info` is part of the object itself and identifies itself fully, so `getsize` is likely not needed anymore. But the same could be asked about `Tree.size` too, since its hash_info also has `size`. So looks like we could use it there as well. WDYT?

Reply:
Though you actually access `Tree.size`, when you are computing the `Tree.hash_info` so they two are coupled. We could inline the current logic to `Tree.digest()` and then use `Tree.hash_info.size` in `Tree.size` but that is a bit different from what you asked since we still need to compute. I'd say this is the urgent fix considering it is slowing down operations, but if you'd like to see that too, I can include.

Reply:
I've dropped the `getsize` completely, and also `Tree.size` as you suggested but now `Tree.size` can't be used with `.digest()` first (or without passing `hash_info=...` in the `__init__`).